include/onnxruntime/core/framework/buffer_deleter.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/framework/allocator.h"
+
+namespace onnxruntime {
+
+// TODO: Do we need this class or is IAllocator::MakeUniquePtr sufficient/better
+class BufferDeleter {
+ public:
```

Comment:
Can we use IAllocator::MakeUniquePtr? Usage is more similar to std::make_unique as it handles the allocation and creating a unique pointer in one call.

Reply:
Will address in a different change